packages/vitest/src/integrations/vi.ts

```diff
@@ -723,7 +723,7 @@ function _mocker(): VitestMocker {
   // @ts-expect-error injected by vite-nide
     ? __vitest_mocker__
     : new Proxy(
-      {},
+      {} as any,
```

Comment:
Just more `any` since typescript 5.8 became smarter :upside_down_face: https://devblogs.microsoft.com/typescript/announcing-typescript-5-8/#granular-checks-for-branches-in-return-expressions